sql/sql_parse.cc

```diff
@@ -5153,9 +5153,36 @@ mysql_execute_command(THD *thd)
       my_ok(thd);
     break;
   case SQLCOM_BACKUP_LOCK:
-    if (check_global_access(thd, RELOAD_ACL))
-      goto error;
-    /* first table is set for lock. For unlock the list is empty */
+    if (check_global_access(thd, RELOAD_ACL, true))
+    {
+      /* In case there is no global privilege, check DB privilege for BACK LOCK.*/
+      if (first_table)
+      {
+        if (check_single_table_access(thd, LOCK_TABLES_ACL, first_table, true))
+        {
+#ifndef NO_EMBEDDED_ACCESS_CHECKS
+          char command[30];
+          get_privilege_desc(command, sizeof(command), RELOAD_ACL|LOCK_TABLES_ACL);
+          my_error(ER_SPECIFIC_ACCESS_DENIED_ERROR, MYF(0), command);
+#endif
+          goto error;
+        }
+      }
+      else // BACKUP UNLOCK
+      {
+        if (!thd->mdl_backup_lock)
+        {
+#ifndef NO_EMBEDDED_ACCESS_CHECKS
+          char command[30];
+          get_privilege_desc(command, sizeof(command), RELOAD_ACL);
+          my_error(ER_SPECIFIC_ACCESS_DENIED_ERROR, MYF(0), command);
```

Comment:
so this is a denied message for everyone, if there isn't a `backup lock` in place saying reload is needed?

Reply:
For `backup unlock`.

Reply:
For `backup unlock`

Reply:
`RELOAD_ACL|LOCK_TABLES_ACL` for the error message.